source/dub/commandline.d

```diff
@@ -1211,6 +1211,8 @@ class GenerateCommand : PackageBuildCommand {
 		if (!gensettings.config.length)
 			gensettings.config = m_defaultConfig;
 		gensettings.runArgs = app_args;
+		// legacy compatibility, default working directory is always CWD
+		gensettings.overrideToolWorkingDirectory = getWorkingDirectory();
```

Comment:
I think `--root` should be respected (in the long run). So reverting the behavior is just to avoid breaking code. What is our path forward ?

Reply:
I think it's sane to keep it like this forever.

Using `workingDirectory "."` you get the behavior that workingDirectory == package directory, which you had with the breaking behavior here. So I think it's fine really